### DISCUSSION TITLE
Handle Vercel Function Termination by Execution time limit

### DISCUSSION THREAD

**Discussion by PonomareVlad** [vercel/community#129] [Help] [answered]
Is there a solution for handling a function interrupt signal like `SIGTERM` ?

Or does execution stop immediately, freezing the application state ?

Using Node 14.x 🙂

**Comment by Pieparker** [comment, marked as answer]
Hey there! Serverless functions are independent and not stateful. From a best solution point-of-view, you should be considering whether serverless functions are the correct tool, if you're looking to make the most of the execution time limit and perform long-running tasks.
